x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/capacity/AutoscalingCapacityWireSerializationTests.java

```diff
@@ -57,7 +59,7 @@ protected AutoscalingCapacity mutateInstance(AutoscalingCapacity instance) {
                     instance.total().memory(),
                     hasAllMetrics && (instance.node() == null || instance.node().processors() == null) && randomBoolean()
                         ? null
-                        : randomIntBetween(1, 64) + Optional.ofNullable(instance.total().processors()).orElse(0f)
+                        : Optional.ofNullable(instance.total().processors()).map(p -> p.plus(randomProcessors())).orElse(randomProcessors())
```

Comment:
I wonder if `p.plus` ensures a different value here? If `randomProcessors` becomes `Double.MIN_VALUE` and `p` is already a value with `Math.ulp(value) > Double.MIN_VALUE`, the resulting `Processors` object could be identical, breaking the test?

Similar comment for other uses of `plus` in this class.

Reply:
`Processors#plus` adds the rounded up values. In the case you present we would be rounding up both processors to the minimal representable processor `1E-5` and the result should be `2E-5`.

Reply:
Makes sense, thanks.